src/test/java/io/confluent/idesidecar/restapi/resources/FlinkDataPlaneProxyResourceTest.java

```diff
@@ -0,0 +1,281 @@
+package io.confluent.idesidecar.restapi.resources;
+
+import static io.confluent.idesidecar.restapi.util.ResourceIOUtil.asJson;
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import com.github.tomakehurst.wiremock.client.WireMock;
+import com.github.tomakehurst.wiremock.matching.EqualToPattern;
+import io.confluent.idesidecar.restapi.connections.CCloudConnectionState;
+import io.confluent.idesidecar.restapi.connections.ConnectionStateManager;
+import io.confluent.idesidecar.restapi.exceptions.ProcessorFailedException;
+import io.confluent.idesidecar.restapi.models.ConnectionSpec.ConnectionType;
+import io.confluent.idesidecar.restapi.processors.Processor;
+import io.confluent.idesidecar.restapi.proxy.FlinkDataPlaneProxyProcessor;
+import io.confluent.idesidecar.restapi.proxy.ProxyContext;
+import io.confluent.idesidecar.restapi.testutil.NoAccessFilterProfile;
+import io.confluent.idesidecar.restapi.util.CCloudTestUtil;
+import io.quarkiverse.wiremock.devservice.ConnectWireMock;
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.TestProfile;
+import io.vertx.core.Future;
+import io.vertx.core.MultiMap;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpMethod;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.core.MediaType;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.CompletionException;
+import java.util.stream.Stream;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+@QuarkusTest
+@TestProfile(NoAccessFilterProfile.class)
+@ConnectWireMock
+class FlinkDataPlaneProxyResourceTest {
+
+  @Inject
+  ConnectionStateManager connectionStateManager;
+
+  @ConfigProperty(name = "quarkus.wiremock.devservices.port")
+  int wireMockPort;
+
```

Comment:
The `FlinkDataPlaneProxyProcessor` can be injected in this class because we've added the `@QuarkusTest` annotation.

```suggestion
@Inject
FlinkDataPlaneProxyProcessor flinkDataPlaneProxyProcessor;
```